sympy/solvers/solveset.py

```diff
@@ -1121,3 +1124,597 @@ def linsolve(system, *symbols):
+
+    if intersections:
+        # If solveset have returned some intersection for any symbol
+        # intersection is Interval or Set
+        result = []
+        for res in result_finiteset:
+            res_copy = res
+            for key_res, value_res in res.items():
+                for key_intersec, value_intersec in intersections.items():
+                    if key_intersec == key_res:
+                        res_copy[key_res] = \
+                            Intersection(FiniteSet(value_res), value_intersec)
+            result.append(res_copy)
+        result_finiteset = result
+
+    if complements:
+        # If solveset have returned some complements for any symbol
+        # complements are Set.
+        result = []
+        for res in result_finiteset:
+            res_copy = res
+            for key_res, value_res in res.items():
+                for key_complement, value_complement in complements.items():
+                    if key_complement == key_res:
+                        res_copy[key_res] = \
+                            FiniteSet(value_res) - value_complement
+            result.append(res_copy)
+        result_finiteset = result
+
+    # convert to ordered tuple
+    result = S.EmptySet
+    for r in result_finiteset:
+        temp = [r[symb] for symb in all_symbols]
+        result += FiniteSet(tuple(temp))
+    return result
+
+
+def nonlinsolve(system, symbols):
+    r"""
+    Solve system of N non linear equations with M variables, which means both
+    under and overdetermined systems are supported. Positive dimensional
+    system is also supported (A system with infinitely many solutions is said
+    to be positive-dimensional). In Positive dimensional system solution will
+    be dependent on at least one symbol. Returns both real solution
+    and complex solution. The possible number of solutions is zero,
+    one or infinite.
+
+    Parameters
+    ==========
+
+    system : list of equations
+        The target system of equations
+    symbols : list of Symbols
+        symbols should be given as a sequence eg. list
+
+    Returns
+    =======
+
+    A FiniteSet of ordered tuple of values of `symbols` for which the `system`
+    has solution. Order of values in the tuple is same as symbols present in
+    the parameter `symbols`.
+
+    Please note that general FiniteSet is unordered, the solution returned
+    here is not simply a FiniteSet of solutions, rather it is a FiniteSet of
+    ordered tuple, i.e. the first & only argument to FiniteSet is a tuple of
+    solutions, which is ordered, & hence the returned solution is ordered.
+
+    Also note that solution could also have been returned as an ordered tuple,
+    FiniteSet is just a wrapper `{}` around the tuple. It has no other
+    significance except for the fact it is just used to maintain a consistent
+    output format throughout the solveset.
+
+    For the given set of Equations, the respective input types
```

Comment:
The examples serve the purpose of familiarising the user with the API.
So, I find this unnecessary .

Instead of this, it will be helpful to state the circumstances under which various exceptions are raised by this function.